File: 0088-merge-sorted-array/0088-merge-sorted-array.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    int* nums1Copy = (int*)malloc(sizeof(int) * m);
    for (int i = 0; i < m; i++) {
        nums1Copy[i] = nums1[i];
    }
    
    int i = 0, j = 0, index = 0;
    while (i < m && j < n) {
        if (nums1Copy[i] <= nums2[j]) {
            nums1[index] = nums1Copy[i];
            i++;
        } else {
            nums1[index] = nums2[j];
            j++;
        }
        index++;
    }
    
    while (i < m) {
        nums1[index] = nums1Copy[i];
        i++;
        index++;
    }
    
    while (j < n) {
        nums1[index] = nums2[j];
        j++;
        index++;
    }
    
    free(nums1Copy);
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
    let index =  m + n - 1;
    let i = m - 1;
    let j = n - 1;
    while(i >= 0 && j >= 0) {
        if(nums1[i] >= nums2[j]){
            nums1[index] = nums1[i];
            i -= 1
        } else {
            nums1[index] = nums2[j];
            j -= 1
        }
        index -= 1
    }
    while(j >= 0){
        nums1[index] = nums2[j]
        index -= 1
        j -= 1
    }    
};